Stripe/UIImage+Stripe.m

```diff
@@ -13,17 +13,32 @@ @implementation UIImage (Stripe)
 - (NSData *)stp_jpegDataWithMaxFileSize:(NSUInteger)maxBytes {
     CGFloat scale = 1.0;
     NSData *imageData = UIImageJPEGRepresentation(self, 0.5);
-    while (imageData.length > maxBytes) {
-        scale = scale * (CGFloat) 0.8;
-        CGSize newImageSize = CGSizeMake(self.size.width * scale,
-                                         self.size.height *scale);
-        UIGraphicsBeginImageContextWithOptions(newImageSize, NO, self.scale);
-        [self drawInRect:CGRectMake(0, 0, newImageSize.width, newImageSize.height)];
-        UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
-        UIGraphicsEndImageContext();
-        imageData = UIImageJPEGRepresentation(newImage, 0.5);
-    }
 
+    // Try something smarter first
+    if (imageData.length > maxBytes) {
+        // Assuming jpeg file size roughly scales linearly with area of the image
+        // which is ~correct (although breaks down at really small file sizes)
+        CGFloat percentSmallerNeeded = (CGFloat) maxBytes / imageData.length;
+
+        // Shrink to a little bit less than we need to try to ensure we're under
+        // (otherwise its likely our first pass will be over the limit due to
+        // compression variance and floating point rounding)
+        scale = scale * sqrt(percentSmallerNeeded - (percentSmallerNeeded * 0.05));
```

Comment:
cool, that makes sense! was just wondering if the 5% came from anywhere in particular